core/bind/core_bind.cpp

```diff
@@ -2098,10 +2098,11 @@ void _Thread::_start_func(void *ud) {
 	memdelete(tud);
 	Callable::CallError ce;
 	const Variant *arg[1] = { &t->userdata };
+	int argc = arg[0]->get_type() != 0;
```

Comment:
This should use `Variant::NIL`, not `0`.

And I'd make the cast from `bool` to `int` explicit:

```suggestion
	int argc = (int)(arg[0]->get_type() != Variant::NIL);
```